test/javascript_literals.coffee

```diff
@@ -1,10 +1,47 @@
-# Javascript Literals
+# JavaScript Literals
 # -------------------
 
-# TODO: refactor javascript literal tests
-# TODO: add indexing and method invocation tests: `[1]`[0] is 1, `function(){}`.call()
+test "inline JavaScript is evaluated", ->
+  eq '\\`', `
+    // Inline JS
+    "\\\`"
+  `
```

Comment:
With my proposed escaping rules, this test needs to be written like this:

```coffee
eq '\\`', `
  // Inline JS
  "\\\\\`"
`
```

or like this:

```coffee
eq '\\a`', `
  // Inline JS
  "\\a\`"
`
```